datajoint/fetch.py

```diff
@@ -182,6 +188,11 @@ def __call__(self, *attrs, offset=None, limit=None, order_by=None, format=None,
                 ret = np.array(ret, dtype=heading.as_dtype)
                 for name in heading:
                     ret[name] = list(map(partial(get, heading[name]), ret[name]))
+
+                    if isinstance(ret[name],(np.ndarray)) and len(ret[name]) > 0:
```

Comment:
use a more efficient way to test whether `ret` is empty

Reply:
Addressed.